Library/Homebrew/dependency.rb

```diff
@@ -38,8 +38,12 @@ def installed?
     to_formula.installed?
   end
 
+  def pinned?
+    to_formula.pinned?
+  end
+
   def satisfied?(inherited_options)
-    installed? && missing_options(inherited_options).empty?
+    (installed? || pinned?) && missing_options(inherited_options).empty?
```

Comment:
This is a bit different to what I think we want. Instead of considering this satisfied it should be considered unsatisfied, refuse to upgrade and print out a message telling the user to `unpin`.